relay-server/server/relayServer.go

```diff
@@ -394,30 +394,46 @@ func (lc *LogClient) WatchMessages() error {
 			kg.Warnf("Failed to receive a message (%s)", lc.server)
 			break
 		}
+		select {
+		case MsgBufferChannel <- res:
+		default:
+		}
+	}
+
+	kg.Print("Stopped watching messages from " + lc.server)
 
-		msg := pb.Message{}
+	return nil
+}
 
-		if err := kl.Clone(*res, &msg); err != nil {
-			kg.Warnf("Failed to clone a message (%v)", *res)
-			continue
-		}
+func (rs *RelayServer) AddMsgFromBuffChan() {
 
-		tel, _ := json.Marshal(msg)
-		fmt.Printf("%s\n", string(tel))
+	for Running {
+		select {
+		case res := <-MsgBufferChannel:
+			msg := pb.Message{}
 
-		MsgLock.RLock()
-		for uid := range MsgStructs {
-			select {
-			case MsgStructs[uid].Broadcast <- (&msg):
-			default:
+			if err := kl.Clone(*res, &msg); err != nil {
+				kg.Warnf("Failed to clone a message (%v)", *res)
+				continue
 			}
-		}
-		MsgLock.RUnlock()
-	}
 
-	kg.Print("Stopped watching messages from " + lc.server)
+			tel, _ := json.Marshal(msg)
+			fmt.Printf("%s\n", string(tel))
```

Comment:
while we are at it, can we have the printf wrapped in an env var check so that we do not overload the kubearmor-relay-server stdout with so much of data. I think this by itself will improve the performance drastically that we might not need such big buffers.

Reply:
It will be better to read and keep the value in a global vars.

I would suggest having these:
```
STDOUT_LOGS=1/0
STDOUT_ALERTS=1/0
STDOUT_MSGS=1/0
```

Reply:
Earlier my assumption was overloading the stdout was causing the relay-server to stop working _( as i was not able to see the number of logs received from kubearmor and sent to client )_

but i dont think that's the case , because earlier I tested for 50 nodes with increased buffer size, the **stdout was overloaded** but that had no impact on number of logs sent, and all logs were being received and sent to client 

the bottle neck was the buffersize as we were receiving burst of 10k log per node and we cannot send that many logs to client fast enough hence we create a buffer.  
Not printing can result in improved throughput as we wont be unmarshling and printing the data but i dont think it'll have any significant impact on buffer size.

Regardless i'll test if removing the print statement and reducing the buffersize has any impact

Reply:
results 
[sheet](https://docs.google.com/spreadsheets/d/1VNb6q7I55V5MW_2N8D_AFcKeOGRIa5Ug4XD0qEkp7A4/edit?usp=sharing)

Under the Print column
No means i wasnt printing the logs
yes meant printing all the logs

#### Note:
in the sheet the CPU and Mem utilization is low for Printing logs compared to not printing is because the metric server stops sending the metric and metric wrnt being updated hence cant find the resource utilization while printing logs
[10:20](https://accuknox.slack.com/archives/D04961TAGNQ/p1681836643687469)

### Conclusion:
#### impact on CPU : 
Not printing does sounds better in this case 
inconclusive in terms of number as metric server stops responding while printing logs

#### impact log buffer:
the buffer is the bottle neck and reducing the size resulted in dropped logs but the throughput increased and we can see the difference between printing and not printing logs is about ~10%